src/components/RepoSubmission.tsx

```diff
@@ -1,13 +1,17 @@
 import { useState } from "react";
+
 import { sendMessage } from "../lib/discord";
 import isValidRepoUrl from "../lib/validateUrl";
 import { ToastTrigger } from "../lib/reactHotToast";
 import useSupabaseAuth from "../hooks/useSupabaseAuth";
 
-const RepoSubmission = () => {
+export declare interface RepoSubmissionProps {
+  isFormOpen: boolean;
+  setIsFormOpen: React.Dispatch<React.SetStateAction<boolean>>;
```

Comment:
this is not correct, if you need to pass the global state of the form being open around, you need to wrap that in a hook rather than pass the state function as props - check out the votes hook for an example 🍕

Reply:
also, typing on the `RepoSubmission` component is not done correctly, it should return a `JSX.Element` like this:

```
const RepoSubmission = ({ isFormOpen, setIsFormOpen }: RepoSubmissionProps): JSX.Element => {
```

Making the above change will underline the return statements, for a good example of a loading/error indicator check out HotRepositories component and its use of `useRepo`

Reply:
that should work if something you click does `handleFormOpen(true)`, you can push the code and I will take a look at it tomorrow :D

Reply:
Great :pizza:

Reply:
Hey @0-vortex  I keep getting this error: `Uncaught TypeError: Cannot read properties of undefined (reading 'slice')   at StackedAvatar (StackedAvatar.tsx:5:19) and (StackedAvatar.tsx:8:19)`

Reply:
One thing that I noticed is contributors props from StackedAvatar goes from -> Array(0) -> Array(30) -> Array(0) -> undefined

Reply:
I removed `useRepoForm` because for some reason it doesn't work. But everything else is ok except [this error](https://github.com/open-sauced/hot/pull/310#discussion_r953140265)